src/components/header/components/submenu-panel/component.jsx

```diff
@@ -0,0 +1,255 @@
+import MyGfwIcon from 'assets/icons/mygfw.svg';
+
+import './styles.scss';
+
+class Header extends PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    apps: PropTypes.array,
+    moreLinks: PropTypes.array,
+    showSubmenu: PropTypes.bool,
+    onClick: PropTypes.func,
+    navMain: PropTypes.array,
+    activeLang: PropTypes.object,
+    languages: PropTypes.array,
+    hideMenu: PropTypes.func,
+    handleLangSelect: PropTypes.func,
+    openContactUsModal: PropTypes.func,
+    loggedIn: PropTypes.bool,
+    setQueryToUrl: PropTypes.func,
+    appUrl: PropTypes.string,
+    pathname: PropTypes.string,
+    NavLinkComponent: PropTypes.oneOfType([ PropTypes.node, PropTypes.func ])
+  };
+
+  state = { search: '' };
+
+  handleSubmit = () => {
+    const { setQueryToUrl, hideMenu } = this.props;
+    if (setQueryToUrl) {
+      setQueryToUrl(this.state.search);
+    } else {
+      window.location.href = `https://www.globalforestwatch.org/search?query=${this.state.search}`;
+    }
+    hideMenu();
+  };
+
+  handleSearchChange = search => {
+    this.setState({ search });
+  };
+
+  handleContactUsOpen = () => {
+    const { openContactUsModal } = this.props;
+    if (openContactUsModal) {
+      openContactUsModal();
+    }
+  };
+
+  render() {
+    const {
+      className,
+      apps,
+      moreLinks,
+      navMain,
+      languages,
+      activeLang,
+      handleLangSelect,
+      loggedIn,
+      hideMenu,
+      NavLinkComponent,
+      appUrl,
+      pathname
+    } = this.props;
+
+    return (
+      <div className={cx('c-submenu-panel', className)}>
+        <div className="submenu-wrapper">
+          <Search
+            className="menu-search"
+            placeholder="Search"
+            onChange={this.handleSearchChange}
+            onSubmit={this.handleSubmit}
+          />
+          <Media lessThan="md-bg" className="menu-section">
+            <ul>
+              {
+                navMain && navMain.map(item => (
+                  <li key={item.label} className="nav-item">
+                    <NavLink
+                      {...item}
+                      pathname={pathname}
+                      appUrl={appUrl}
+                      NavLinkComponent={NavLinkComponent}
+                    >
+                      {item.label}
+                    </NavLink>
+                  </li>
+                  ))
+              }
+              <li className="nav-item">
+                <NavLink
+                  href="/my-gfw"
+                  pathname={pathname}
+                  appUrl={appUrl}
+                  NavLinkComponent={NavLinkComponent}
+                >
+                  My GFW
+                  <MyGfwIcon
+                    className={cx('my-gfw-icon', { 'logged-in': loggedIn })}
+                  />
+                </NavLink>
+              </li>
+            </ul>
+          </Media>
+          <Media lessThan="md-bg" className="menu-section">
```

Comment:
maybe we could take it to the next level of declarativity and move these <Media lessThan="..."> to its own components `<Desktop />`, `<Tablet />`, `<Mobile />` etc

Reply:
also because `<Media lessThan="md-bg">` is already in a couple of places

Reply:
Yeah I can do that. Sensible idea.

Reply:
Hmm after reviewing this it might be over engineering. To cover the cases we have here we would need at least the following components:
`export const Mobile = (props) => <Media lessThan="md">{props.children}</Media>;
export const MobileWide = (props) => <Media lessThan="md-bg">{props.children}</Media>;
export const DesktopNarrow = (props) => <Media greaterThanOrEqual="md-bg">{props.children}</Media>;
export const Desktop = (props) => <Media greaterThanOrEqual="md">{props.children}</Media>;`

As we have a unqiue breakpoint for the header at 850 px specifically because of the visibility of menu item at 860pxs. I suggest adding Just `<Desktop />` and `<Mobile />` for now and leaving the custom ones in the header.